server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java

```diff
@@ -653,16 +657,6 @@ public void clusterStateProcessed(ClusterState oldState, ClusterState newState)
                     assertEquals(executor.batches.get(), executor.published.get());
                 }
             }
-
-            // assert the correct number of clusterStateProcessed events were triggered
-            for (Map.Entry<String, AtomicInteger> entry : processedStates.entrySet()) {
-                assertThat(submittedTasksPerThread, hasKey(entry.getKey()));
-                assertEquals(
-                    "not all tasks submitted by " + entry.getKey() + " received a processed event",
-                    entry.getValue().get(),
-                    submittedTasksPerThread.get(entry.getKey()).get()
-                );
-            }
```

Comment:
Asserting global publication count instead of per thread.

Reply:
:+1: makes sense